src/App.tsx

```diff
@@ -1,26 +1,289 @@
+        setLoading(true);
+        setProcessedIs(existing => [...existing, todo.id]);
+        deleteTodo(todo.id)
+          .then(() =>
+            setTodos(existing =>
+              existing.filter(current => current.id !== todo.id),
+            ),
+          )
+          .catch(() => setErrorMessage('Unable to delete a todo'))
+          .finally(() => {
+            setLoading(false);
+            setProcessedIs(existing => existing.filter(id => id !== todo.id));
+          });
+      }
+    });
+  };
+
+  const handleStatusUpdate = (id: number) => {
+    setProcessedIs(existing => [...existing, id]);
+
+    const changeItem = todos.find(todo => todo.id === id);
+
+    if (changeItem) {
+      const toUpdate = { completed: !changeItem.completed };
+
+      updateTodo(id, toUpdate)
+        .then(() => {
+          setTodos(existing =>
+            existing.map(el =>
+              el.id === id ? { ...el, completed: toUpdate.completed } : el,
+            ),
+          );
+        })
+        .catch(() => setErrorMessage('Unable to update a todo'))
+        .finally(() => {
+          setProcessedIs([]);
+        });
+    }
+  };
+
+  const handleTotalStatusUpdate = () => {
+    if (!areAllTodosCompleted) {
+      todos.forEach(todo => {
+        if (!todo.completed) {
+          setLoading(true);
+          setProcessedIs(existing => [...existing, todo.id]);
+          const toUpdate = { completed: true };
+
+          updateTodo(todo.id, toUpdate)
+            .then(() => {
+              setTodos(existing =>
+                existing.map(el =>
+                  el.id === todo.id
+                    ? { ...el, completed: toUpdate.completed }
+                    : el,
+                ),
+              );
+            })
+            .catch(() => setErrorMessage('Unable to update a todo'))
+            .finally(() => {
+              setLoading(false);
+              setProcessedIs(existing => existing.filter(id => id !== todo.id));
+            });
+        }
+      });
+    }
+
+    if (areAllTodosCompleted) {
+      todos.forEach(todo => {
+        setLoading(true);
+        setProcessedIs(existing => [...existing, todo.id]);
+        const toUpdate = { completed: false };
+
+        updateTodo(todo.id, toUpdate)
+          .then(() => {
+            setTodos(existing =>
+              existing.map(el =>
+                el.id === todo.id
+                  ? { ...el, completed: toUpdate.completed }
+                  : el,
+              ),
+            );
+          })
+          .catch(() => setErrorMessage('Unable to update a todo'))
+          .finally(() => {
+            setLoading(false);
+            setProcessedIs(existing => existing.filter(id => id !== todo.id));
+          });
+      });
+    }
+  };
+
+  const handleTitleUpdate = (
+    id: number,
+    newTitle: string,
+    setTitleBeingUpdated: React.Dispatch<React.SetStateAction<boolean>>,
+    isTitleChanged: boolean,
+    e?: React.FormEvent<HTMLFormElement>,
```

Comment:
do not pass it as an arguments

Instead, in TodoItem component create a separate function handler `onSubmit`. In this function, you should apply local logic (event.preventDefault, input focus, etc.) and just call this function with todo id and new title

Apply the same approach for other functions as well